src/Symfony/Bridge/Doctrine/Listener/DoctrineConnectionListener.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Listener;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class DoctrineConnectionListener implements EventSubscriberInterface
+{
+    public function __construct(private readonly \ArrayObject $connectionTiming, private ContainerInterface $container, private readonly array $timingbyConnection)
+    {
+    }
+
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        $connectionTiming = $this->connectionTiming->getIterator();
+
+        if (!$connectionTiming) {
+            return;
+        }
+
+        foreach ($connectionTiming as $key => $value) {
+            $connectionName = sprintf('doctrine.dbal.%s_connection', $value['name']);
```

Comment:
this is not the connection name. It looks like it is the service id.

I would suggest keeping `timingbyConnection` indexed by name